### PR TITLE
Fix performance problem with toLocale by switching to lazy val

### PR DESCRIPTION
**Pull request by miracle2121**
A simple fix to avoid creating a separate Locale object every time.

### CODE REVIEW

File: framework/src/play/src/main/scala/play/api/i18n/Messages.scala [hunk 1]
```diff
@@ -29,12 +29,12 @@ import scala.io.Codec
  */
 case class Lang(language: String, country: String = "") {
 
+  private lazy val locale = Option(country).filterNot(_.isEmpty).map(c => new java.util.Locale(language, c)).getOrElse(new java.util.Locale(language))
```
Comment (cchantep) [inline framework/src/play/src/main/scala/play/api/i18n/Messages.scala]: Would prefer a single `lazy val toLocale`

**Comment by jroper** [comment]
Looks good. @miracle2121 would you mind squashing your commits?  For information on why/how, see [here](https://playframework.com/documentation/2.3.x/WorkingWithGit).

**Comment by miracle2121** [comment]
Commits squashed.

**Comment by jroper** [comment]
Thanks @miracle2121.

**Pull request merged**